qa/rpc-tests/sc_cert_listsinceblock.py

```diff
@@ -0,0 +1,381 @@
+from test_framework.util import assert_false, assert_true, assert_equal, initialize_chain_clean, \
+    start_nodes, start_node, sync_blocks, sync_mempools, connect_nodes_bi, mark_logs, \
+    get_epoch_data, swap_bytes, get_spendable 
+from test_framework.mc_test.mc_test import *
+import os
+import pprint
+from decimal import Decimal
+from test_framework.mininode import COIN
+import json
+import time
+
+NUMB_OF_NODES = 3
+DEBUG_MODE = 1
+EPOCH_LENGTH = 20
+FT_SC_FEE = Decimal('0')
+MBTR_SC_FEE = Decimal('0')
+CERT_FEE = Decimal("0.00025")
+CUSTOM_FEE_RATE_ZAT_PER_BYTE = Decimal('20.0')
+CUSTOM_FEE_RATE_ZEN_PER_KBYTE = CUSTOM_FEE_RATE_ZAT_PER_BYTE/COIN*1000
+
+class ScCertListsinceblock(BitcoinTestFramework):
+    alert_filename = None
+
+    def setup_chain(self, split=False):
+        print("Initializing test directory " + self.options.tmpdir)
+        initialize_chain_clean(self.options.tmpdir, NUMB_OF_NODES)
+        self.alert_filename = os.path.join(self.options.tmpdir, "alert.txt")
+        with open(self.alert_filename, 'w'):
+            pass  # Just open then close to create zero-length file
+
+    def setup_network(self, split=False):
+
+        self.nodes = start_nodes(NUMB_OF_NODES, self.options.tmpdir,
+            extra_args=[['-logtimemicros=1', '-debug=cert', '-debug=sc', '-debug=py']]*NUMB_OF_NODES)
+
+        connect_nodes_bi(self.nodes, 0, 1)
+        connect_nodes_bi(self.nodes, 1, 2)
+
+        self.is_network_split = split
+        self.sync_all()
+
+    def run_test(self):
+        '''
+        Test that the JSON result of the rpc command listsinceblock includes a matured certificate backward transfer even when:
+        1) the input block range specified does not contain the block where such certificate has been mined.
+        2) The certificate matures in one of the blocks of the specified range
+        '''
+
+        mark_logs("Node 1 generates 2 block",self.nodes,DEBUG_MODE)
+        self.nodes[1].generate(2)
+        self.sync_all()
+
+        mark_logs("Node 0 generates {} block".format(MINIMAL_SC_HEIGHT-2),self.nodes,DEBUG_MODE)
+        self.nodes[0].generate(MINIMAL_SC_HEIGHT-2)
+        self.sync_all()
+
+        #generate wCertVk and constant
+        mcTest = CertTestUtils(self.options.tmpdir, self.options.srcdir)
+        vk = mcTest.generate_params('sc1')
+        constant = generate_random_field_element_hex()
+
+        # create SC
+        #------------------------------------------------------------------------------------------------------------
+        cmdInput = {
+            'toaddress': "abcd", 'amount': 20.0, 'wCertVk': vk, 'withdrawalEpochLength': EPOCH_LENGTH,'constant': constant}
+
+        mark_logs("\nNode 1 create SC", self.nodes, DEBUG_MODE)
+        try:
+            res = self.nodes[1].sc_create(cmdInput)
+            tx =   res['txid']
+            scid = res['scid']
+            pprint.pprint(res)
+            self.sync_all()
+        except JSONRPCException, e:
```

Comment:
For future compatibility with Python 3, we could use `as` in place of `,` for the exception.

```suggestion
        except JSONRPCException as e:
```